### ISSUE TITLE
BartTokenizer prepare_seq2seq_batch() does not return decoder_input_ids,  decoder_attention_mask as per document after passing tgt_texts

### ISSUE THREAD

**Issue by MojammelHossain** [huggingface/transformers#7846]
I am trying to train a seq2seq model using BartModel. As per BartTokenizer documentation if I pass tgt_texts then it should return decoder_attention_mask and decoder_input_ids please check the attachment for clarity. 
![image](https://user-images.githubusercontent.com/47693507/96223213-5d10af00-0faf-11eb-9253-5985990f15f9.png)
But I am only getting input_ids, attention_mask and labels.
![image](https://user-images.githubusercontent.com/47693507/96223257-71ed4280-0faf-11eb-9c8f-b26b7af8dd9a.png)

**Comment by freespirit** [comment]
I am facing the same issue and I noticed that the method indeed returns  the `["input_ids"]` of `tgt_texts` as labels. I think I could easily fix this to return both `input_ids` and `attention_mask` of `tgt_texts` but I noticed the same pattern in other seq2seq models, like T5. I am not sure what's the proper solution but if it is similar to what I suggest, than I'd be happy to make a pull request.

**Comment by freespirit** [comment]
I think https://github.com/huggingface/transformers/pull/6654/ and https://github.com/huggingface/transformers/issues/6624 are related - the PR changed `decoder_input_ids` to `labels`. Probably the documentation should be changed but I have to get more familiar with the respective issue and PR to be sure.

**Comment by MojammelHossain** [comment]
Thanks for the feedback @freespirit. Hopefully, they will update the documentation as it is a little bit confusing. But what I found that the modeling_bart.py file already handles the problem. _prepare_bart_decoder_inputs() and shift_tokens_right() solving that if I am not wrong. But I think I have to go deeper for understanding which I am trying to.
![image](https://user-images.githubusercontent.com/47693507/97541324-2c893600-19ef-11eb-9a3e-3a1057c70d71.png)
![image](https://user-images.githubusercontent.com/47693507/97541362-3c087f00-19ef-11eb-948b-842378cf0039.png)

**Comment by LysandreJik** [comment]
Pinging @sshleifer for advice

**Comment by sshleifer** [comment]
@MojammelHossain is correct, the docs are wrong. 
The correct usage is to allow `_prepare_bart_decoder_inputs` to make `decoder_input_ids` and `decoder_attention_mask` for you. For training, you only need to pass the 3 keys returned by `prepare_seq2seq_batch`.
